components/url/lib.rs

```diff
@@ -163,6 +165,12 @@ impl ServoUrl {
 
 impl fmt::Display for ServoUrl {
     fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        if self.0.to_string().len() > 40 {
+            let hasher = DefaultHasher::new();
+            let truncated : String = self.0.to_string().chars().take(40).collect();
+            let result = format!("{}{}",truncated,hasher.finish().to_string());
```

Comment:
I suggested the `{}... ({})` format for a reason :) I think the output will be more readable that way.